pkg/reconcile/pipeline/context/impl.go

```diff
@@ -118,15 +121,44 @@ func (i *impl) Services() ([]pipeline.Service, error) {
 func (i *impl) Applications() ([]pipeline.Application, error) {
 	if i.applications == nil {
 		if ref := i.serviceBinding.Spec.Application; ref != nil {
-			gvr, err := i.typeLookup.ResourceForReferable(ref)
-			if err != nil {
-				return nil, err
+			if i.serviceBinding.Spec.Application.Name != "" &&
+				i.serviceBinding.Spec.Application.LabelSelector != nil &&
+				i.serviceBinding.Spec.Application.LabelSelector.MatchLabels != nil {
+				return nil, errs.New("Both label selector and name for application should be used")
```

Comment:
you mean - "should not be used"?

Reply:
Done. 5326c6f674cb869fdaa8242bd4cc454fb72941cc